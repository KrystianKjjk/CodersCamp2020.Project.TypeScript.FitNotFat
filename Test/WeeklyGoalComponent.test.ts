import {
  generateWeeklyGoalComponent
} from '../Src/UIComponents/WeeklyGoalComponent/WeeklyGoalComponent';
import { WeeklyGoal } from '../Models/WeeklyGoal.model';

describe('Tests for Weekly goal component', () => {
  test('should render weekly goal component', () => {
    const weeklyGoalComponent=generateWeeklyGoalComponent('1',()=>{});  
    expect(weeklyGoalComponent).toMatchSnapshot();
  });

  test('should save selected goal',()=>{
    const onClickFunction=jest.fn();
    const weeklyGoalComponent=generateWeeklyGoalComponent('1',onClickFunction);  
    const firstOption:HTMLDivElement=weeklyGoalComponent.querySelector('.select-weekly-goal .option');
    firstOption.click();
    const saveButton:HTMLButtonElement=weeklyGoalComponent.querySelector('.button-save');
    saveButton.click();
    expect(onClickFunction).toHaveBeenCalledWith(WeeklyGoal.Gain);
})
});
