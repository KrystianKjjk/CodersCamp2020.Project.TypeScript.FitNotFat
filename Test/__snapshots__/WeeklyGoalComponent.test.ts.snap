// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Weekly goal component should render weekly goal component 1`] = `
<div
  class="tile"
>
  <div
    class="weekly-goal-component"
  >
    <p
      class="tile-title"
    >
      Weekly goal
    </p>
    <p
      class="paragraph-week"
    >
      1
    </p>
    <div
      class="container select-weekly-goal"
      style="width: 300px;"
    >
      <p
        class="select-header"
      >
        What is your weekly goal:
      </p>
      <div
        class="select-box"
      >
        <span>
          Gain weight
        </span>
        <span>
          
    
          <svg
            fill="currentColor"
            height="16"
            viewBox="0 0 16 16"
            width="16"
            xmlns="http://www.w3.org/2000/svg"
          >
            
        
            <path
              d="M1.646 4.646a.5.5 0 0 1 .708 0L8 10.293l5.646-5.647a.5.5 0 0 1 .708.708l-6 6a.5.5 0 0 1-.708 0l-6-6a.5.5 0 0 1 0-.708z"
              fill-rule="evenodd"
            />
            
    
          </svg>
        </span>
      </div>
      <div
        class="options"
        style="display: none;"
      >
        <div
          class="option selected"
          data-key="Gain"
        >
          Gain weight
        </div>
        <div
          class="option"
          data-key="Lose"
        >
          Lose weight
        </div>
        <div
          class="option"
          data-key="Keep"
        >
          Keep weight
        </div>
      </div>
    </div>
    <button
      class="button button-save"
      style="background-color: white; color: rgb(218, 27, 54);"
      type="button"
    />
  </div>
</div>
`;
